Makefile

```diff
@@ -2,25 +2,13 @@ VERSION = $(shell cat VERSION)
 LIBRARY_PREFIX = pam_watchid
 LIBRARY_NAME = $(LIBRARY_PREFIX).so
 DESTINATION = /usr/local/lib/pam
-TARGET = apple-darwin$(shell uname -r)
 PAM_FILE_BASE = /etc/pam.d/sudo
 PAM_TEXT = auth sufficient $(LIBRARY_NAME)
 PAM_TID_TEXT = auth       sufficient     pam_tid.so
 
-# Determine if the macOS Sequoia SDK or later is available.
-DEFINES =
-# Due to the different ways in which the CLT and Xcode structure their SDK paths, one of the following will always be an empty string depending on what is configured by xcode-select. 
-CLT_SDK_MAJOR_VER = $(shell xcrun --sdk macosx --show-sdk-path | xargs readlink -f | xargs basename | sed 's/MacOSX//' | cut -d. -f1)
-XCODE_SDK_MAJOR_VER = $(shell xcrun --sdk macosx --show-sdk-path | xargs basename | sed 's/MacOSX//' | cut -d. -f1)
-SDK_REQUIRED_MAJOR_VER = 15
-ifeq "$(SDK_REQUIRED_MAJOR_VER)" "$(word 1, $(sort $(SDK_REQUIRED_MAJOR_VER) $(XCODE_SDK_MAJOR_VER) $(CLT_SDK_MAJOR_VER)))"
-	DEFINES += -DSEQUOIASDK
-endif
-
 all:
-	swiftc watchid-pam-extension.swift $(DEFINES) -o $(LIBRARY_PREFIX)_x86_64.so -target x86_64-$(TARGET) -emit-library
-	swiftc watchid-pam-extension.swift $(DEFINES) -o $(LIBRARY_PREFIX)_arm64.so -target arm64-$(TARGET) -emit-library
-	lipo -create $(LIBRARY_PREFIX)_arm64.so $(LIBRARY_PREFIX)_x86_64.so -output $(LIBRARY_NAME)
+	swift build -c release --arch x86_64 --arch arm64
+	mv .build/apple/Products/Release/libpam-watchid.dylib $(LIBRARY_NAME)
```

Comment:
Found a couple of issues when building with the CLT.

Support for building swift packages with the CLT (especially in older versions of macOS) seems to be quite poor. In Sonoma and later, builds are significantly longer than they previously were (about double in CI and 10x on my machine). In Ventura and earlier, they outright don't seem to work.

Likewise,

> is there any reason why we're ... not going straight to something like:
> ```
> swift build -c release --arch x86_64 --arch arm64
> ```

Turns out there is a reason not to do it like that. For whatever reason, multiarch builds don't seem to work with the CLT, instead failing with:

```
error: xcbuild executable at '/Library/Developer/SharedFrameworks/XCBuild.framework/Versions/A/Support/xcbuild' does not exist or is not executable
```

Unless I've completely overlooked something to get CLT builds working, I think it would probably be best if we continue to use the old build process when the CLT is in use, and leave the swift package for full Xcode installations and builds for Nixpkgs.

```suggestion
ifeq ($(shell [[ '$(shell xcode-select -p)' == '/Library/Developer/CommandLineTools' ]] && echo true),true)
# Legacy build
# For CLT due to poor support for building swift packages.
# Swift packages do work in macOS Sonoma and later with the CLT, but are an order of magnitude slower than Xcode. 
	swiftc Sources/pam-watchid/pam_watchid.swift -o $(LIBRARY_PREFIX)_x86_64.so -target x86_64-$(TARGET) -emit-library
	swiftc Sources/pam-watchid/pam_watchid.swift -o $(LIBRARY_PREFIX)_arm64.so -target arm64-$(TARGET) -emit-library
	lipo -create $(LIBRARY_PREFIX)_arm64.so $(LIBRARY_PREFIX)_x86_64.so -output $(LIBRARY_NAME)
else
# Swift Package Manager build
	swift build -c release --arch x86_64 --arch arm64
	mv .build/apple/Products/Release/libpam-watchid.dylib $(LIBRARY_NAME)
endif
```

`TARGET` should also be reinstated to how you had it in your previous commit.

Reply:
The `Makefile` actually doesn't matter at all for the Nix build, because it's going to build directly with SPM . As such, I'm happy to have the `Makefile` unconditionally use `swiftc` directly if you'd like. Let me know whether you'd prefer that and I'll make the change

Reply:
I considered that but I’d rather still be testing the Swift Package Manager build somewhere. It’s also a much cleaner way of building in places where it’s supported.